samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -438,11 +442,48 @@ public void publishJobModelVersion(String oldVersion, String newVersion) {
         "(actual data version after update = " + stat.getVersion() + ")");
   }
 
+  // validate Zk protocol currently used is the same as in this participant
+  private void validateZkVersion() {
+    String rootPath = keyBuilder.getRootPath();
+    if (!zkClient.exists(rootPath)) {
```

Comment:
actually this check is for version validation. It is just that there could be two different scenarios:
1. RootPath already exists - then we need to read and validate the version.
2. RootPath doesn't exist - then we need to create one and write the version into it.